### PR TITLE
Feat/pupil 631/programme page

### PR DESCRIPTION
**Pull request by kimon-satan**
### User Story

**I want to be able to choose available programme factors for a given subjecy**

### Implementation details

pupils/years/YEAR-SLUG/subjects/SUBJECT_SLUG/programmes

- implement with an SPA pattern
- buttons for initial page / links for final page
- back button is link for initial page / button if there was a previous page

https://www.figma.com/file/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?type=design&node-id=3993%3A45782&mode=dev

Notes for Jim:
- I have refactored getStaticPaths and getStaticProps methods into a separate folder see src/pages-helpers . If you like this approach I can encourage others to build up this folder rather than copy pasting into pages

Just a skeleton no need for final components

#### The full experience
pupils/beta/programmes/combined-science-secondary-year-11/options

### only tiers
https://deploy-preview-2395--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/maths-secondary-year-10/options

### only examboards
https://deploy-preview-2395--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/english-secondary-year-11/options

#### Compare legacy and non legacy programmes
https://deploy-preview-2395--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/maths-secondary-year-10/options-l
https://deploy-preview-2395--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/maths-secondary-year-10/options

#### no programme factors available
https://deploy-preview-2395--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/english-secondary-year-8/options

### Acceptance criteria

- [x]  served from pupils/beta/pupils/programmes/maths-secondary-year-10/options
- [x]  presents only available programme factors
- [x]  presents examboards followed by tiers
- [x]  final page links to correct unit listing page
- [x]  back button - pupils/beta/pupils/years/YEAR_SLUG/subjects
- [x]  accepts a query string for a programme factor already selected eg. ?examboard=aqa
    - [x]  this means that only the tier needs to be selected
- [x]  if no programme factors redirect to unit lisitng page

### CODE REVIEW

File: src/pages/pupils/beta/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx [hunk 1]
```diff
@@ -5,14 +5,20 @@ import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
 import { LessonListingBrowseData } from "@/node-lib/curriculum-api-2023/queries/pupilLessonListing/pupilLessonListing.schema";
 import getPageProps from "@/node-lib/getPageProps";
 import { resolveOakHref } from "@/common-lib/urls";
-import { URLParams } from "@/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons";
+import { getStaticPaths as getStaticPathsTemplate } from "@/pages-helpers/get-static-paths";
 
-export { getStaticPaths } from "@/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons";
```
Comment (kimon-satan) [inline src/pages/pupils/beta/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx:10]: This was a pretty dangerous anti-pattern which was written in response to sonarCloud's complaints about code duplication on pages.

**Review by jimCresswell — APPROVED** [review]
Wow, that is a chunky piece of work. It looks good to me from an architecture point of view. I haven't tested the user journey.

File: src/pages/pupils/beta/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx [hunk 2]
```diff
@@ -5,14 +5,20 @@ import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
 import { LessonListingBrowseData } from "@/node-lib/curriculum-api-2023/queries/pupilLessonListing/pupilLessonListing.schema";
 import getPageProps from "@/node-lib/getPageProps";
 import { resolveOakHref } from "@/common-lib/urls";
-import { URLParams } from "@/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons";
```
Comment (kimon-satan) [inline src/pages/pupils/beta/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx:8]: We should discourage generic names like URLParams as they lead people into anti-patterns like this one.

**Review by benprotheroe — APPROVED** [review]

**Review by RBrownsett — APPROVED** [review]
Tested scenarios:
-Tier only (Maths Year 10)
-Exam Board Only (History year 10)
-No programme factors

**Pull request merged**